api/src/main/java/jakarta/data/repository/Pagination.java

```diff
@@ -118,24 +151,73 @@ public List<Sort> sorts() {
 
     @Override
     public Pageable newPage(long pageNumber) {
-        return new Pagination(pageNumber, size, sorts);
+        return new Pagination(pageNumber, size, sorts, mode, cursor);
     }
 
     @Override
     public Pageable newSize(int maxPageSize) {
-        return new Pagination(page, maxPageSize, sorts);
+        return new Pagination(page, maxPageSize, sorts, mode, cursor);
     }
 
     @Override
     public Pageable sortBy(Iterable<Sort> sorts) {
         List<Sort> sortList = sorts == null
                 ? Collections.emptyList()
                 : StreamSupport.stream(sorts.spliterator(), false).collect(Collectors.toUnmodifiableList());
-        return new Pagination(page, size, sortList);
+        return new Pagination(page, size, sortList, mode, cursor);
     }
 
     @Override
     public Pageable sortBy(Sort... sorts) {
-        return new Pagination(page, size, sorts == null ? Collections.emptyList() : List.of(sorts));
+        return new Pagination(page, size, sorts == null ? Collections.emptyList() : List.of(sorts), mode, cursor);
+    }
+
+    /**
+     * Built-in implementation of Cursor.
+     */
+    static final class CursorImpl implements Cursor {
```

Comment:
It is not meaningful, once it is a default implementation.

```suggestion
static final class DefaultCursor implements Cursor {
```

Reply:
Good input, I'll get it renamed, but will need to do so independently of the Commit suggestion due additional usage by the unit test class.